pkg/agent/datamodel/types.go

```diff
@@ -18,6 +18,43 @@ import (
 	"github.com/Azure/go-autorest/autorest/to"
 )
 
+// WindowsProfile represents the windows parameters passed to the cluster
+type WindowsProfile struct {
+	AdminUsername                 string                `json:"adminUsername"`
+	AdminPassword                 string                `json:"adminPassword" conform:"redact"`
+	CSIProxyURL                   string                `json:"csiProxyURL,omitempty"`
+	EnableCSIProxy                *bool                 `json:"enableCSIProxy,omitempty"`
+	ImageRef                      *api.ImageReference   `json:"imageReference,omitempty"`
+	ImageVersion                  string                `json:"imageVersion"`
+	ProvisioningScriptsPackageURL string                `json:"provisioningScriptsPackageURL,omitempty"`
+	WindowsImageSourceURL         string                `json:"windowsImageSourceURL"`
+	WindowsPublisher              string                `json:"windowsPublisher"`
+	WindowsOffer                  string                `json:"windowsOffer"`
+	WindowsSku                    string                `json:"windowsSku"`
+	WindowsDockerVersion          string                `json:"windowsDockerVersion"`
+	Secrets                       []api.KeyVaultSecrets `json:"secrets,omitempty"`
+	SSHEnabled                    *bool                 `json:"sshEnabled,omitempty"`
+	EnableAutomaticUpdates        *bool                 `json:"enableAutomaticUpdates,omitempty"`
+	IsCredentialAutoGenerated     *bool                 `json:"isCredentialAutoGenerated,omitempty"`
+	EnableAHUB                    *bool                 `json:"enableAHUB,omitempty"`
+	WindowsPauseImageURL          string                `json:"windowsPauseImageURL"`
+	AlwaysPullWindowsPauseImage   *bool                 `json:"alwaysPullWindowsPauseImage,omitempty"`
+}
+
```

Comment:
FYI @AbelHu  For future windows cse migration to agentbaker.

Reply:
Does this mean that AKS RP will start to use WindowsProfile under agentbaker?

Reply:
AKS RP and AgentBaker currently have their own data models. RP converts its data model to AgentBaker's before calling AgentBaker's functions.